### PR TITLE
refactor: unify offer id and swap

### PR DESCRIPTION
**Pull request by h4sh3d**
Offer was using `Uuid` and swaps `SwapId`. We want to make ids compatible to allow transition from `trade => swap` in the node (also see #296). This PR introduce a wrapper type against `Uuid` that implements strict encode and strict decode for ease of use in the node and use this new `Uuid` for offers and swaps in core.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -115,3 +121,90 @@ pub enum Error {
...
+#[display(inner)]
+pub struct Uuid(uuid::Uuid);
+
+impl Uuid {
+    /// Creates a new uuid for trades and swaps.
+    pub fn new() -> Self {
+        Self(uuid::Uuid::new_v4())
+    }
+
+    /// Creates a new random uuid, same as `Self::new()`.
```
Comment (TheCharlatan) [inline src/lib.rs:138]: What is the purpose of this extra function?

**Review by TheCharlatan — CHANGES_REQUESTED** [review]
There is a selection use case where the swap id type is useful. How about wrapping the Uuid struct in a SwapId, respectively OfferId struct and implement from/into for both and vice-versa?

File: src/lib.rs (hunk 1 above)
Comment (h4sh3d) [reply src/lib.rs:138]: To keep the API compatible with what we have before: `SwapId::random()`

**Comment by h4sh3d** [comment]
> There is a selection use case where the swap id type is useful. How about wrapping the Uuid struct in a SwapId, respectively OfferId struct and implement from/into for both and vice-versa?

You prefer having two different structs to possibly pattern match them, correct?

We can keep the current `Uuid` and use two type defs, I think that would work.

**Comment by TheCharlatan** [comment]
> You prefer having two different structs to possibly pattern match them, correct?
> 
> We can keep the current `Uuid` and use two type defs, I think that would work.

I don't think a pattern match will apply here, but I still feel like they should not be mixed up with each other without an explicit `.into()` call.

**Comment by h4sh3d** [comment]
@TheCharlatan changes done

**Comment by h4sh3d** [comment]
Waiting on #296 to be finalized and merge to then come back on that one.

**Review by TheCharlatan — APPROVED** [review]

**Pull request merged**